src/raft/raft.rs

```diff
@@ -614,24 +616,31 @@ impl<T: Storage> Raft<T> {
 
     // tick_election is run by followers and candidates after self.election_timeout.
     // TODO: revoke pub when there is a better way to test.
-    pub fn tick_election(&mut self) {
+    // Return true to indicate that handle_raft_ready needs to be called.
+    pub fn tick_election(&mut self) -> bool {
         self.election_elapsed += 1;
-        if self.promotable() && self.pass_election_timeout() {
-            self.election_elapsed = 0;
-            let m = new_message(INVALID_ID, MessageType::MsgHup, Some(self.id));
-            self.step(m).is_ok();
+        if !self.pass_election_timeout() || !self.promotable() {
+            return false;
         }
+
+        self.election_elapsed = 0;
+        let m = new_message(INVALID_ID, MessageType::MsgHup, Some(self.id));
+        self.step(m).is_ok();
+        true
     }
 
     // tick_heartbeat is run by leaders to send a MsgBeat after self.heartbeat_timeout.
-    fn tick_heartbeat(&mut self) {
+    // Return true to indicate that handle_raft_ready needs to be called.
```

Comment:
I'm not sure whether it's a good idea to write comment like this mentioning the interface of a outer module which calls `raw_node.tick()`.